subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/inMemory/arrays/PersistedBooleanArray.java

```diff
@@ -9,6 +9,7 @@
 import java.util.Arrays;
 import java.util.Iterator;
 
+@SuppressWarnings({"PMD.ArrayIsStoredDirectly", "PMD.MethodReturnsInternalArray"})
```

Comment:
why are we suppressing these here?

Reply:
code was written 2008, does it really make sense to rewrite it, and clone the bytes? if something is working for so long imo it is better to focus on newly introduced vulnerabilites.

Reply:
see above

Reply:
same here